api/v1alpha1/ext_auth_types.go

```diff
@@ -106,3 +116,27 @@ type BodyToExtAuth struct {
 	// +kubebuilder:validation:Minimum=1
 	MaxRequestBytes uint32 `json:"maxRequestBytes"`
 }
+
+// ExtAuthAccessibleMetadata defines options related to the sending of dynamic metadata to and from the
+// external authorization service.
+type ExtAuthAccessibleMetadata struct {
+	// Namespaces are metadata namespaces that are sent to the external authorization server as context.
+	//
+	// +optional
+	Namespaces []string `json:"namespaces,omitempty"`
+
+	// TypedNamespaces are typed metadata namespaces that are sent to the external authorization server as context.
+	//
+	// +optional
+	TypedNamespaces []string `json:"typedNamespaces,omitempty"`
```

Comment:
Do we need both typed an untyped? I think that for ext-proc, we only support untyped at the moment.

Reply:
Hmm, I prefer if we can support both, as Confluent makes use of https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_filters/set_metadata_filter to set some typed metadata. I see other OSS filters also set typed metadata (such as `gcp_authn`) as well as a few of Confluent's custom filters.

Supporting typed metadata essentially comes for free, though I understand the user experience might be confusing. WDYT @guydc?